lib/pure/math.nim

```diff
@@ -941,6 +941,44 @@ func euclMod*[T: SomeNumber](x, y: T): T {.since: (1, 5, 1).} =
   if result < 0:
     result += abs(y)
 
+func ceilDiv*[T: SomeInteger](x, y: T): T {.inline, since: (1, 5, 1).} =
+  ## Ceil division is conceptually defined as `ceil(x / y)`.
+  ##
+  ## This is different from the `system.div <system.html#div,int,int>`_
+  ## operator, which is defined as `trunc(x / y)`.
+  ## That is, `div` rounds towards `0` and `ceilDiv` rounds up.
+  ##
+  ## **See also:**
+  ## * `system.div proc <system.html#div,int,int>`_ for integer division
+  ## * `floorDiv func <#floorDiv,T,T>`_ for integer division with rounds down.
+  ## * `fastCeilDiv func <#fastCeilDiv,T,T>`_ for faster integer division.
+  runnableExamples:
+    doAssert ceilDiv( 12,  3) ==  4
+    doAssert ceilDiv( 13,  3) ==  5
+    doAssert ceilDiv(-13,  3) == -4
+    doAssert ceilDiv( 13, -3) == -4
+    doAssert ceilDiv(-13, -3) ==  5
```

Comment:
Examples should use `assert` by convention instead I think.

Reply:
Really? Most of runnableExamples in this module use `doAssert`.

Reply:
don't follow the local convention, follow the convention in contributing.html; otherwise old conventions persist